packages/keystone/src/fields/types/float/views/index.tsx

```diff
@@ -11,22 +12,141 @@ import {
   FieldProps,
 } from '../../../../types';
 import { CellLink, CellContainer } from '../../../../admin-ui/components';
+import { useFormattedInput } from '../../integer/views/utils';
 
-export const Field = ({ field, value, onChange, autoFocus }: FieldProps<typeof controller>) => (
-  <FieldContainer>
-    <FieldLabel htmlFor={field.path}>{field.label}</FieldLabel>
-    {onChange ? (
+type Validation = {
+  min?: number;
+  max?: number;
+  isRequired?: boolean;
+};
+
+type Value =
+  | { kind: 'update'; initial: number | null; value: string | number | null }
+  | { kind: 'create'; value: string | number | null };
+
+function validate(value: Value, validation: Validation, label: string) {
+  const val = value.value;
+
+  // if we recieve null initially on the item view and the current value is null,
+  // we should always allow saving it because:
+  // - the value might be null in the database and we don't want to prevent saving the whole item because of that
+  // - we might have null because of an access control error
+  if (value.kind === 'update' && value.initial === null && val === null) {
+    return undefined;
+  }
+
+  if (value.kind === 'create' && value.value === null) {
+    return undefined;
+  }
+
+  if (validation.isRequired && val === null) {
+    return `${label} is required`;
+  }
+
+  if (typeof val === 'number') {
+    if (!Number.isFinite(val)) {
+      return `${label} must be finite`;
+    }
+    if (validation?.min !== undefined && val < validation.min) {
+      return `${label} must be greater than or equal to ${validation.min}`;
+    }
+    if (validation?.max !== undefined && val > validation?.max) {
+      return `${label} must be less than or equal to ${validation.max}`;
+    }
+  }
+
+  return undefined;
+}
+
+function FloatInput({
+  value,
+  onChange,
+  id,
+  autoFocus,
+  forceValidation,
+  validationMessage,
+  placeholder,
+}: {
+  id: string;
+  autoFocus?: boolean;
+  value: number | string | null;
+  onChange: (value: number | string | null) => void;
+  forceValidation?: boolean;
+  validationMessage?: string;
+  placeholder?: string;
+}) {
+  const [hasBlurred, setHasBlurred] = useState(false);
+  const props = useFormattedInput<number | null>(
+    {
+      format: value => (value === null ? '' : value.toString()),
+      parse: raw => {
+        raw = raw.trim();
+        if (raw === '') {
+          return null;
+        }
+        if (/^[+-]?\d+\.?\d?$/.test(raw)) {
```

Comment:
Can you remove regex check, there's other syntax that should work here like `5e200`?